### PR TITLE
refactor parsing of commands, implement providers as subcommands, fix #43

### LINKED ISSUE #43: provider should be verified before proceeding
Running this:

```bash
[vagrant@compose ~]$ henge xyz
file "docker-compose.yml" not found
```

This should return some output like, `unknown provider: xyz` or `unknown argument: xyz` and not look for a docker-compose.yml file.

### PR DESCRIPTION
**Pull request by containscafeine**
@kadel @surajssd I have removed the requirement for `pkg/transformers/main.go` and created subcommands for openshift and kubernetes, so most of the things are being taken care of by cobra now.

Let me know what do you think of this approach so I can polish this and send the final PR.

Thanks.

### CODE REVIEW

File: pkg/cmd/kubernetes.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+			err = kubernetes.PrintList(list, vals)
+
+			if err != nil {
+				fmt.Fprintln(os.Stderr, err.Error())
+				os.Exit(1)
+			}
+		},
+	}
+
+	cmd.Flags().StringSliceVarP(&vals.Files, "files", "f", []string{"docker-compose.yml"}, "Provide docker-compose files, comma separated.")
```
Comment (surajssd) [inline pkg/cmd/kubernetes.go]: Can we have this repetition of flag addition in one place?

File: pkg/cmd/root.go [hunk 2]
```diff
@@ -39,18 +37,13 @@ func Execute() (*types.CmdValues, error) {
 		Short:   "Henge converts the docker compose file to various orchestration providers' artifacts.",
 		Long:    cliLong,
 		Example: example,
```
Comment (surajssd) [inline pkg/cmd/root.go]: We can add `Run` here and show help?

File: pkg/cmd/kubernetes.go (hunk 1 above)
Comment (containscafeine) [inline pkg/cmd/kubernetes.go]: @surajssd these can be, but that would also make both `henge -f ___ openshift` and `henge openshift -f ___` possible. I think it should only be the latter, wdyt?

File: pkg/cmd/root.go (hunk 2 above)
Comment (surajssd) [inline pkg/cmd/root.go]: yes, just typing `henge` it does show help :)

**Comment by surajssd** [comment]
LGTM!

**Comment by surajssd** [comment]
@kadel once you LGTM, we can merge it!

**Comment by kadel** [comment]
Only one thing. If you do it like this than this part  no longer executed: https://github.com/containscafeine/henge/blob/7d922b4047cf7a53ea5caa793c737ce2e997223c/henge.go#L20,L24

Or more exactly it is executed, but only after everything else, and only if there was no error before ;-)

**Comment by surajssd** [comment]
nice catch, i missed it somehow!

**Comment by containscafeine** [comment]
@surajssd @kadel, gotcha, thanks. How about now?

**Comment by kadel** [comment]
:+1: LGTM
We can squash it using magic merge button ;-)

**Comment by containscafeine** [comment]
@kadel @surajssd, squashed. Thanks :)

**Comment by surajssd** [comment]
cool, merging! Thanks @containscafeine !

**Pull request merged**